sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DistinctTest.java

```diff
@@ -251,4 +253,64 @@ public void testTriggeredDistinctRepresentativeValues() {
         KV.of(1, "k1"), KV.of(2, "k2"), KV.of(3, "k3"));
     triggeredDistinctRepresentativePipeline.run();
   }
+
+  /**
+   * Regression test: when all values are emitted by a speculative trigger, caused a null KV
+   * when the on-time firing occurred.
+   */
+  @Test
+  @Category({ValidatesRunner.class, UsesTestStream.class})
```

Comment:
Does this need to be a `ValidatesRunner` test, or is `NeedsRunner` sufficient? Same for the other one.

Reply:
Good call. Made it `NeedsRunner`.